psi4/src/psi4/libfock/v.cc

```diff
@@ -679,11 +679,8 @@ void VBase::set_grac_shift(double grac_shift) {
     if (!grac_initialized_) {
         double grac_alpha = options_.get_double("DFT_GRAC_ALPHA");
         double grac_beta = options_.get_double("DFT_GRAC_BETA");
-        std::shared_ptr<Functional> grac_x_func = static_cast<std::shared_ptr<Functional>>(
-            new LibXCFunctional(options_.get_str("DFT_GRAC_X_FUNC"), functional_->is_unpolarized()));
-
-        std::shared_ptr<Functional> grac_c_func = static_cast<std::shared_ptr<Functional>>(
-            new LibXCFunctional(options_.get_str("DFT_GRAC_C_FUNC"), functional_->is_unpolarized()));
+        auto grac_x_func = std::make_shared<LibXCFunctional>(options_.get_str("DFT_GRAC_X_FUNC"), functional_->is_unpolarized());
+        auto grac_c_func = std::make_shared<LibXCFunctional>(options_.get_str("DFT_GRAC_C_FUNC"), functional_->is_unpolarized());
```

Comment:
isn't this going to make LHS a `LibXCFunctional`, not a `Functional`?

Reply:
Yes, but that's not a problem. The old version of the code _explicitly_ upcasts to the parent object. The new version _implicitly_ upcasts when these objects are passed as arguments to the `set_grac_*_functional` methods, according to their method signatures.